File: Intrif/pesadelo.cpp
#include <iostream>
#include <math.h>
#include <tgmath.h>
#include <iomanip>

using namespace std;

// #define PI 3.1415926535897932384

// 452617


int main()
{

    long double PI = 3.14159265358979323846L;

    int r;
    cin >> r;

    if (r < 1 || r > 100)
    {
        return 0;
    }

    long double azinho = 36.869897645L;

    long double theta = 73.739795266608263933829256231553122024909673218277166042005784900L;

    long double alpha = (r / 2.0L);
    long double x = (r * 2.0L);
    long double y = (5.0L * alpha);
    long double z = (3.0L * alpha);

    long double ab = (pow(x, 2.0L) / y);

    long double h = sqrt((pow(r, 2.0L)) - ((pow(ab, 2.0L)) / 4.0L));

    /*

    55.051554320147
    55.051554320147

    */

    cout << fixed << setprecision(12);

    long double a1 = ((ab * h) / 2.0L);

    long double a2 = ((theta * PI * (pow(r, 2.0))) / 360.0L);

    long double areaTotal = a1 + a2;

    cout << areaTotal << endl;

    return 0;
}

File: Intrif/pesadelo.py
from decimal import Decimal, getcontext
import math

# Definir a precisão decimal desejada (12 casas decimais)
getcontext().prec = 14  # Ajuste para 14 para garantir 12 casas decimais após a vírgula

PI = Decimal('3.14159265358979323846')

r = int(input())

if r < 1 or r > 100:
    exit()

azinho = Decimal('36.869897645')
theta = Decimal('73.739795266608263933829256231553122024909673218277166042005784900')

theta = round(theta, 12)

alpha = Decimal(r) / Decimal('2.0')
x = Decimal(r) * Decimal('2.0')
y = Decimal('5.0') * alpha
z = Decimal('3.0') * alpha

ab = x**2 / y
h = (r**2 - ab**2 / Decimal('4.0')).sqrt()

a1 = (ab * h) / Decimal('2.0')
a2 = (theta * PI * (r**2)) / Decimal('360.0')

areaTotal = a1 + a2

